programs/drift/src/controller/position.rs

```diff
@@ -115,21 +134,23 @@ pub fn update_position_and_market(
             (new_quote_entry_amount, new_quote_break_even_amount, 0_i64)
         }
         PositionUpdateType::Reduce | PositionUpdateType::Close => {
-            let new_quote_entry_amount = position.quote_entry_amount.safe_sub(
-                position
-                    .quote_entry_amount
-                    .cast::<i128>()?
-                    .safe_mul(delta.base_asset_amount.abs().cast()?)?
-                    .safe_div(position.base_asset_amount.abs().cast()?)?
-                    .cast()?,
-            )?;
+            let current_base_i128 = position.get_current_base_with_remainder_abs()?;
+            let delta_base_i128 = delta.get_delta_base_with_remainder_abs()?;
+            let ddd = position
```

Comment:
would be good to have a better name or you can just make it match the format before where this calculation happens inside the safe sub